src/main/java/seedu/address/model/charge/ChargeDate.java

```diff
@@ -0,0 +1,115 @@
+package seedu.address.model.charge;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Represents the month-year of a Pet to be charged in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class ChargeDate {
+    public static final String MESSAGE_CONSTRAINTS = "Charge date should be formatted as MM-yyyy!";
+
+    /**
+     * Regex to match charge date format.
+     * eg: "03-2022".
+     */
+    public static final String VALIDATION_REGEX = "\\d{2}-\\d{4}\\s*";
+
+    private final int month;
+    private final int year;
+
+    /**
+     * Constructs an {@code ChargeDate}.
+     *
+     * @param chargeDate A valid charge date.
+     */
+    public ChargeDate(String chargeDate) {
+        requireNonNull(chargeDate);
+        checkArgument(isValidChargeDate(chargeDate), MESSAGE_CONSTRAINTS);
+        String[] monthYear = chargeDate.trim().split("-");
+        int month = Integer.valueOf(monthYear[0]);
+        int year = Integer.valueOf(monthYear[1]);
+        this.month = month;
+        this.year = year;
+    }
+
```

Comment:
Is there a reason why you want to use `int` and the following methods in this class instead of parsing the string and format it as a `LocalDate` so that extracting the Date and Year would be easier?

Reply:
Right at first I didn't want LocalDate because as far as I know, it has to be instantiated with a day. I have found a YearMonth class that I will use instead, thanks for pointing this out!